packages/orchestrator/internal/sandbox/checks.go

```diff
@@ -65,3 +84,55 @@ func (s *Sandbox) Healthcheck(ctx context.Context, alwaysReport bool) {
 		return
 	}
 }
+
+func (s *Sandbox) GetMetrics(ctx context.Context) (SandboxMetrics, error) {
+	address := fmt.Sprintf("http://%s:%d/metrics", s.slot.HostIP(), consts.DefaultEnvdServerPort)
+
+	request, err := http.NewRequestWithContext(ctx, "GET", address, nil)
+	if err != nil {
+		return SandboxMetrics{}, err
+	}
+
+	response, err := httpClient.Do(request)
+	if err != nil {
+		return SandboxMetrics{}, err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		err = fmt.Errorf("unexpected status code: %d", response.StatusCode)
+		return SandboxMetrics{}, err
+	}
+
+	var metrics SandboxMetrics
+	err = json.NewDecoder(response.Body).Decode(&metrics)
+	if err != nil {
+		return SandboxMetrics{}, err
+	}
+
+	return metrics, nil
+}
+
+func (s *Sandbox) LogMetrics(ctx context.Context) {
+	if isMetricsSupported(s.Sandbox.EnvdVersion) {
+		metrics, err := s.GetMetrics(ctx)
+		if err != nil {
+			s.Logger.Warnf("failed to get metrics: %s", err)
+		} else {
+			s.Logger.CPUPct(metrics.CPUPercent)
+			s.Logger.MemMB(metrics.MemMB)
+		}
+	}
+}
+
+func isMetricsSupported(envdVersion string) bool {
+	if !semver.IsValid("v" + envdVersion) {
+		return false
+	}
+
+	if len(envdVersion) > 0 && envdVersion[0] != 'v' {
+		envdVersion = "v" + envdVersion
+	}
```

Comment:
is the order correct? If it has `v` prefix, it will already fail on the first condition, won't it?